### PR TITLE
feat:  Add emoji/emoticon extraction to single word feature [PT-188413705]

### PR DESCRIPTION
**Pull request by dougmartin**
This expands the existing wordTokenizer function to extract emojis and emoticons.

### CODE REVIEW

File: src/lib/emoticons.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+// and copying them into emoticons.txt and then extracting the initial characters up to the
+// first space using: awk '{print $1}' emoticons.txt
+
+const rawEmoticons = `
+$:
+%)
+%-)
+&-:
+&:
+(
```
Comment (tealefristoe) [inline src/lib/emoticons.ts:12]: This one seems like it might cause problems.

**Review by tealefristoe — APPROVED** [review]
Looks great 👍

File: src/lib/emoticons.ts (hunk 1 above)
Comment (dougmartin) [reply src/lib/emoticons.ts:12]: It is escaped on line 252.

Comment (dougmartin) [reply src/lib/emoticons.ts:12]: Ah - I see, you mean just the bare paren.  You are right, I'll remove that.

Comment (dougmartin) [reply src/lib/emoticons.ts]: @tealefristoe I removed the bare open paren in follow on commit: https://github.com/concord-consortium/storyq-codap-plugin/pull/19/commits/05d55ee9a63311b6ea8d59d5b98c0fbdf9fbb7b7

**Pull request merged**